crates/holochain_websocket/src/lib.rs

```diff
@@ -688,13 +729,59 @@ impl WebsocketListener {
         Ok(Self {
             config,
             access_control,
-            listener,
+            listener: Box::new(listener),
+        })
+    }
+
+    /// Bind a new websocket listener on the same port using a v4 and a v6 socket.
+    pub async fn dual_bind(
+        config: Arc<WebsocketConfig>,
+        addr_v4: impl Into<SocketAddr>,
+        addr_v6: impl Into<SocketAddr>,
+    ) -> Result<Self> {
```

Comment:
It is possible to configure a system with no v4 loopback device, and likewise with no v6 loopback device. In the case where only one bind is successful, do we still want this function to work, just returning only a single result in the local_addrs vec?

Reply:
Yes, you're right that is possible and we may as well handle it cleanly. I don't know how I can test this on CI but I have updated the implementation and run it with IPv6 disabled on my machine and it works nicely.

I didn't use a `vec` because the implementation gets fairly grim if you need a dynamic `accept`. The `select!` is much cleaner and easier. What I can do though is fall back to the default constructor which just binds to a single address in the case that an interface is disabled. This seems like a clean solution to me :)